elasticsearch_dsl/utils.py

```diff
@@ -233,7 +245,7 @@ def __init__(cls, name, bases, attrs):
         # skip for DslBase
         if not hasattr(cls, "_type_shortcut"):
             return
-        if cls.name is None:
```

Comment:
This is (I think) the only logic change I've made. The `name` attribute defaults to `None` in base classes, and is set to the name of the entity in each subclass. To avoid having to type this as `Optional[str]` and then having to deal with null checks everywhere, I change the default to an empty string.